### ISSUE TITLE
Expose GDAL driver metadata

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/pyogrio#103]
Would we want something like this in fiona: https://github.com/Toblerity/Fiona/pull/950 ?

It would allow to check the extension of a driver, the open options, whether the driver supports certain data types, etc (but it would probably be mostly copy/paste from fiona ..)

**Comment by theroggy** [comment]
Sounds like it is at least somewhat linked to #81 and #71

**Comment by jorisvandenbossche** [comment]
> linked to https://github.com/geopandas/pyogrio/issues/81

Yes, if we have this, it could indeed be used for #81. But I think it's also certainly fine to already do something about #81 without this feature (this issue is also meant for user facing API, while for the tests we can hard-code a bunch of things on the short term)

**Comment by theroggy** [comment]
I wonder if it would be possible to get more driver metadata into gdal... this way it is available centrally instead of having copies of this metadata in every python library + in every other programming language.

**Comment by theroggy** [comment]
> Which additional information are you thinking about? (the current GDAL metadata already exposes quite a lot: extensions, which data types are supported, whether nulls are supported, ..)

Ah, OK, I wasn't aware of that. In that case, ignore my remark :-).

**Comment by jorisvandenbossche** [comment]
I started using the GDAL driver metadata internally in https://github.com/geopandas/pyogrio/pull/189, but without publicly exposing it for now.
